libs/linear/Data/Linear/LVect.idr

```diff
@@ -48,3 +53,35 @@ export
 Consumable a => Consumable (LVect n a) where
   consume [] = ()
   consume (x :: xs) = x `seq` consume xs
+
+||| map a linear vector
+export
+map : (0 f : a -@ b) -> {auto 1 fns : n `Copies` f} -> LVect n a -@ LVect n b
+map f {fns=[]} [] = []
+map f {fns=(f :: fs)} (x :: xs) = f x :: map f {fns=fs} xs
+
+export
+length : Consumable a => LVect n a -@ LNat
+length [] = Zero
+length (x :: xs) = let () = consume x in Succ (length xs)
+
+export
+foldl : {0 f : acc -@ elem -@ acc} -> n `Copies` f -@ acc -@ (LVect n elem) -@ acc
```

Comment:
The convention is a bit inconsistent between `map f {fns}` and `foldl {f} fns`.
Can we rely on auto search for these or should we uniformly take the `fns` explicitly?

Reply:
Oh right, auto-search works fine for both, we should use `foldl f {fns}`, will fix